Alpaca-Finance-Agents/AllocPoint-Change-Alert/src/agent.ts

```diff
@@ -0,0 +1,113 @@
+import { 
+  Finding, 
+  HandleTransaction, 
+  TransactionEvent, 
+  FindingSeverity, 
+  FindingType 
+} from 'forta-agent';
+import { 
+  decodeParameters
+} from "forta-agent-tools";
+import {
+  POOL_CONTROLLERS_ADDRESSES,
+  MDEX_PCS_POOLS
+} from "./pools";
+
+const mdexSetAbi: string = "function set(uint256 _pid, uint256 _allocPoint, bool _withUpdate)";
+
+const pcsQueueTxnAbi: string = "event QueueTransaction(bytes32 indexed txHash, address indexed target, uint value, string signature, bytes data, uint eta)";
+const pcsSetFuncSig: string = "set(uint256,uint256,bool)";
+
+let timelockController: string;
+let boardroomController: string;
+
+const createFinding = (
+  posId: number,
+  allocPoint: number,
+  withUpdate: boolean,
+  target: string | null
+  ): Finding => {
+  return Finding.fromObject({
+    name: "AllocPoint Change Event",
+    description: "Pool's alloc point queued for update.",
+    alertId: "ALPACA-6",
+    severity: FindingSeverity.Info,
+    type: FindingType.Info,
+    metadata:{
+      positionId: posId.toString(),
+      allocPoint: allocPoint.toString(),
+      withUpdate: withUpdate.toString(),
+      target: target || "N/A" // target ARGUMENT COULD POTENTIALLY BE null
+    },
+  })
+}
+
+export function provideHandleTransaction(
+  pools: Map<string, Map<number, string>>,
+  poolControllers: string[]
+): HandleTransaction {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+    const queueTxnEvents = txEvent.filterLog(pcsQueueTxnAbi);
+    // TODO: FIND DEPLOYED BoardRoomMDX TO FILTER BY ITS ADDRESS 
+    const mdexSetCalls = txEvent.filterFunction(mdexSetAbi);
+
+    if(queueTxnEvents.length > 0) {
```

Comment:
you can actually remove this `if`. When the length is `0` the `for` won't do any iteration.